src/philips-hue.coffee

```diff
@@ -52,144 +54,158 @@
 # Author:
 #   kingbin - chris.blazek@gmail.com
 
+hue = require("node-hue-api")
+HueApi = hue.HueApi
+lightState = hue.lightState
 
 module.exports = (robot) ->
   base_url = process.env.PHILIPS_HUE_IP
   hash  = process.env.PHILIPS_HUE_HASH
+  api = new HueApi(base_url, hash)
+  state = lightState.create();
+
+  # GROUP COMMANDS
+  robot.respond /hue (?:ls )?groups/i, (msg) ->
+    api.groups (err, groups) ->
+      return handleError msg, err if err
+      robot.logger.debug groups
+      msg.send "Light groups:"
+      for group in groups
+        if group.id == '0'
+          msg.send "- #{group.id}: '#{group.name}' (All)"
+        else
+          msg.send "- #{group.id}: '#{group.name}' (#{group.lights.join(', ')})"
 
-
-# If you have more than the initial 3 lights, add them to the 'all' array below
-  fake_groups =
-    1: [1]
-    2: [2]
-    3: [3]
-    all: [1,2,3]
-
-# GROUP COMMANDS
-  robot.respond /hue groups/i, (msg) ->
-    url = "http://#{base_url}/api/#{hash}/groups"
-    getGenInfo msg, url, (responseText) ->
-      msg.send "Groups: " + responseText
-
-# FAKE GROUP COMMANDS
   robot.respond /hue group (\w+)=(\[((\d)+,)*((\d)+)\])/i, (msg) ->
-    msg.send "setting " + msg.match[1] + "=" +msg.match[2]
-    fake_groups[msg.match[1]] = JSON.parse(msg.match[2])
-
-  robot.respond /hue rm group (\w+)/i, (msg) ->
-    msg.send "removing " + msg.match[1]
-    delete fake_groups[msg.match[1]]
-
-  robot.respond /hue ls groups/i, (msg) ->
-    for key of fake_groups
-      msg.send key + " = [" + fake_groups[key] + "]"
-
-# LIGHT COMMANDS
+    group_name = msg.match[1]
+    group_lights = JSON.parse(msg.match[2])
+    msg.send "Setting #{group_name} to #{group_lights.join(', ')}"
+    api.createGroup group_name, group_lights, (err, response) ->
+      return handleError msg, err if err
+      robot.logger.debug response
+      msg.send "Group created!"
+
+  robot.respond /hue rm group (\d)/i, (msg) ->
+    group_id = msg.match[1]
+    msg.send "Deleting #{group_id}"
+    api.deleteGroup group_id, (err, response) ->
+      return handleError msg, err if err
+      robot.logger.debug response
+      msg.send "Group deleted!"
+
+  # LIGHT COMMANDS
   robot.respond /hue lights/i, (msg) ->
-    light = msg.match[1]
-    url = "http://#{base_url}/api/#{hash}/lights"
-    getGenInfo msg, url, (responseText) ->
-      msg.send responseText
+    api.lights (err, lights) ->
+      return handleError msg, err if err
+      robot.logger.debug lights
+      msg.send "Connected hue lights:"
+      for light in lights.lights
+        msg.send "- #{light.id}: '#{light.name}'"
 
   robot.respond /hue light (.*)/i, (msg) ->
```

Comment:
So, on [this line](https://github.com/kingbin/hubot-philipshue/pull/8/files#diff-0e5b7366c920d484c553c76ca7c225f4R109) I'm still sending in the full response object to `robot.logger.debug`, which will show up if a user runs it from the shell (e.g. `./bin/hubot -n Hubot`) with an environment variable of `export HUBOT_LOG_LEVEL=debug` set.

My thought here is to pick out a few things we know they would want to use (hence the reference to #1), but have the rest of it already available for output if they were truly trying to debug something with a particular bulb via Hubot. I considered that a compromise between verbose output and the basic information.

I do think a template for output (internationalization or otherwise) would be a good feature, though. Particularly if they wanted to maintain the sass-factor of the robot telling them exactly where they could screw that bulb. :smiling_imp: